### PR TITLE
Fix port binding commands issues.

### PR DESCRIPTION
**Pull request by JoeyZhang-Microsemi**
The output of command port-bind-info does not display strings for
bind state and port binding commands use wrong condition for error
check.

### CODE REVIEW

File: cli/main.c [hunk 1]
```diff
@@ -919,10 +919,32 @@ static int arbitration_set(int argc, char **argv)
 	return 0;
 }
 
+static void print_bind_info(struct switchtec_bind_status_out status)
+{
+	int result = status.bind_state & 0x0F;
+	int state = (status.bind_state & 0xF0) >> 4;
+
+	switch (result) {
+	case 0:
```
Comment (lsgunth) [inline cli/main.c]: We should probably have an enum for these case values somewhere.

File: cli/main.c [hunk 2]
```diff
@@ -940,23 +962,19 @@ static int port_bind_info(int argc, char **argv)
 
 	ret = switchtec_bind_info(cfg.dev, &bind_status, cfg.phy_port);
 
-	if (ret < 0) {
-		printf("bind_info_error: %f", ret);
+	if (ret > 0) {
```
Comment (lsgunth) [inline cli/main.c]: You still need to handle the case where switchtec_bind_info() returns less than zero (indicating an error submitting the command or something like that.

File: cli/main.c [hunk 3]
```diff
@@ -940,23 +962,19 @@ static int port_bind_info(int argc, char **argv)
...
+		printf("bind_info_error: %d\n", ret);
```
Comment (lsgunth) [inline cli/main.c]: Probably should also be using switchtec_perror()

File: cli/main.c (hunk 2 above)
Comment (JoeyZhang-Microsemi) [reply cli/main.c]: The function switchtec_bind_info()/switchtec_bind()/switchtec_unbind() cannot return a value that is less then zero. If the return value is bigger than zero, it indicates a error.

File: cli/main.c (hunk 3 above)
Comment (JoeyZhang-Microsemi) [reply cli/main.c]: Because same error codes were used by different MRPC commands and have different meaning. so  we just display the error code now.

File: cli/main.c (hunk 1 above)
Comment (JoeyZhang-Microsemi) [reply cli/main.c]: I agree it. Thanks

File: cli/main.c (hunk 2 above)
Comment (lsgunth) [reply cli/main.c]: That's wrong. switchtec_bind_info() calls switchtec_cmd() and returns the result. switchtec_cmd() can always return negative numbers.

File: cli/main.c (hunk 3 above)
Comment (lsgunth) [reply cli/main.c]: Well returning a number is incredibly user unfriendly. You should fix that and decode the number into a useful error message. If you can use switchtec_perror() that would be great, if not you should create something else. switchtec_bind_perror() or something.

File: lib/switchtec.c [hunk 4]
```diff
@@ -589,26 +616,18 @@ float switchtec_die_temp(struct switchtec_dev *dev)
...
 {
-	int ret;
-
 	struct switchtec_bind_status_in sub_cmd_id = {
 		.sub_cmd = MRPC_PORT_INFO,
 		.phys_port_id = phy_port
 	};
 
-	ret = switchtec_cmd(dev, MRPC_PORTPARTP2P, &sub_cmd_id,
+	return switchtec_cmd(dev, MRPC_PORTPARTP2P, &sub_cmd_id,
```
Comment (lsgunth) [inline lib/switchtec.c]: Nice clean up, but can this be another commit?

**Comment by JoeyZhang-Microsemi** [comment]
Thanks for the review. I has split the commit. 

Thanks
Joey

**Comment by wesleywesley** [comment]
Following enum overlap the system error number:

ERR_PHYC_PORT_ARDY_BIND = 0x00000001, 
ERR_LOGC_PORT_ARDY_BIND = 0x00000002, 
…
ERR_BIND_CHECK_FAIL = 0x0000000d,

Which lead to false prompt in case of non-port-bind related commands by following code in switchtec_strerror function:

	case ERR_PHYC_PORT_ARDY_BIND:
		msg = "Physical port already bound"; break; 
	case ERR_LOGC_PORT_ARDY_BIND: 
		msg = "Logical bridge instance already bound"; break; 
        …
	case ERR_BIND_CHECK_FAIL: 
		msg = "Port bind checking failed"; break;
